test/unit/config/is-gutenberg-plugin.js

```diff
@@ -1,6 +1,34 @@
+/*
+Package.json defines IS_GUTENBERG_PLUGIN as follows:
+
+	"config": {
+		"IS_GUTENBERG_PLUGIN": true
+	}
+
+Unfortunately, in here, it gets stringified like this:
+
+  "true" === process.env.npm_package_config_IS_GUTENBERG_PLUGIN
+
+Webpack replaces the occurrences of IS_GUTENBERG_PLUGIN with the boolean
+value, and the unit tests need to do the same.
+
+This code below converts the string value to its boolean counterpart,
+or preserves the original string in case it contains some unexpected
+value.
+*/
+const isPluginRaw = process.env.npm_package_config_IS_GUTENBERG_PLUGIN;
+let isPlugin;
+if ( isPluginRaw === 'true' ) {
+	isPlugin = true;
+} else if ( isPluginRaw === 'false' ) {
+	isPlugin = false;
+} else {
+	isPlugin = isPluginRaw;
+}
+
 global.process.env = {
 	...global.process.env,
 	// Inject the `IS_GUTENBERG_PLUGIN` global, used for feature flagging.
 	// eslint-disable-next-line @wordpress/is-gutenberg-plugin
-	IS_GUTENBERG_PLUGIN: process.env.npm_package_config_IS_GUTENBERG_PLUGIN,
+	IS_GUTENBERG_PLUGIN: isPlugin,
```

Comment:
<img width="130" alt="Screenshot 2022-04-29 at 07 09 58" src="https://user-images.githubusercontent.com/699132/165888172-7b6afb7d-a04b-4ea3-b586-7ceab866c1a4.png">

Maybe `IS_GUTENBERG_PLUGIN: String( process.env.npm_package_config_IS_GUTENBERG_PLUGIN ) === 'true',` would do the trick?

Reply:
That's what I've done at first, but then I thought it will break if the flag gets an unexpected value such as `"maybe"` one day. But now that I've written my thinking down, I want to say YAGNI :-) Let's do it as you proposed